integration-tests/src/test/java/com/datastax/oss/driver/core/pool/AdvancedShardAwarenessIT.java

```diff
@@ -0,0 +1,287 @@
+import com.datastax.oss.driver.api.core.session.Session;
+import com.datastax.oss.driver.api.testinfra.CassandraSkip;
+import com.datastax.oss.driver.api.testinfra.ccm.CustomCcmRule;
+import com.datastax.oss.driver.api.testinfra.session.SessionUtils;
+import com.datastax.oss.driver.internal.core.pool.ChannelPool;
+import com.datastax.oss.driver.internal.core.util.concurrent.CompletableFutures;
+import com.datastax.oss.driver.internal.core.util.concurrent.Reconnection;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.util.concurrent.Uninterruptibles;
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.slf4j.LoggerFactory;
+
+@CassandraSkip(description = "Advanced shard awareness relies on ScyllaDB's shard aware port")
+public class AdvancedShardAwarenessIT {
+
+  @ClassRule
+  public static final CustomCcmRule CCM_RULE =
+      CustomCcmRule.builder().withNodes(2).withJvmArgs("--smp=3").build();
+
+  public static ch.qos.logback.classic.Logger channelPoolLogger =
+      (ch.qos.logback.classic.Logger) LoggerFactory.getLogger(ChannelPool.class);
+  public static ch.qos.logback.classic.Logger reconnectionLogger =
+      (ch.qos.logback.classic.Logger) LoggerFactory.getLogger(Reconnection.class);
+  ListAppender<ILoggingEvent> appender;
+  Level originalLevelChannelPool;
+  Level originalLevelReconnection;
+  private final Pattern shardMismatchPattern =
+      Pattern.compile(".*r configuration of shard aware port.*");
+  private final Pattern reconnectionPattern =
+      Pattern.compile(".*Scheduling next reconnection in.*");
+  Set<Pattern> forbiddenOccurences = ImmutableSet.of(shardMismatchPattern, reconnectionPattern);
+
+  @Before
+  public void startCapturingLogs() {
+    originalLevelChannelPool = channelPoolLogger.getLevel();
+    originalLevelReconnection = reconnectionLogger.getLevel();
+    channelPoolLogger.setLevel(Level.DEBUG);
+    reconnectionLogger.setLevel(Level.DEBUG);
+    appender = new ListAppender<>();
+    appender.setContext(
+        ((Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME)).getLoggerContext());
+    channelPoolLogger.addAppender(appender);
+    reconnectionLogger.addAppender(appender);
+    appender.list.clear();
+    appender.start();
+  }
+
+  @After
+  public void stopCapturingLogs() {
+    appender.stop();
+    appender.list.clear();
+    channelPoolLogger.setLevel(originalLevelChannelPool);
+    reconnectionLogger.setLevel(originalLevelReconnection);
+    channelPoolLogger.detachAppender(appender);
+    reconnectionLogger.detachAppender(appender);
+  }
+
+  @Test
+  public void should_initialize_all_channels() {
```

Comment:
Can you please add the test to connect to non-shard-aware port and check that it does not target shard-aware port for connection pool

Reply:
I'm not sure I understand what this is supposed to check exactly.
I think `should_see_mismatched_shard` already does that. It targets non-shard-aware port, and has new option enabled. It's verified by seeing the logs about the mismatch.